File: scripts/find_english_texts.py
import os, re, json

PAT = re.compile(r'["\']((?:[^"\']|\\.){3,250})["\']')
RUSSIAN = re.compile(r'[\u0400-\u04FF]')

def is_english_text(s):
    s = s.strip()
    if len(s) < 3:
        return False
    if not re.search(r'[A-Za-z]', s):
        return False
    if RUSSIAN.search(s):
        return False
    # Skip strings that are just code tokens/imports/paths/JSON keys
    if re.match(r'^[A-Za-z0-9_\-\.\/:@#\u0026\[\]\(\)\|]+$|^\$\{.*\}$|^\w+$', s):
        return False
    return True

targets = [
    r"D:\АГЕНТ\ДЖАРВИС\src\components\repo",
    r"D:\АГЕНТ\ДЖАРВИС\src\components\os",
    r"D:\АГЕНТ\ДЖАРВИС\src\components\layout",
    r"D:\АГЕНТ\ДЖАРВИС\src\components\projects",
    r"D:\АГЕНТ\ДЖАРВИС\src\components\sidebar",
    r"D:\АГЕНТ\ДЖАРВИС\src\components\settings",
    r"D:\АГЕНТ\ДЖАРВИС\src\app\repos",
    r"D:\АГЕНТ\ДЖАРВИС\src\app\page.tsx",
    r"D:\АГЕНТ\ДЖАРВИС\src\lib",
    r"D:\АГЕНТ\ДЖАРВИС\src\services",
]

results = []
for t in targets:
    if os.path.isfile(t):
        files = [t]
    else:
        files = []
        for root, dirs, fnames in os.walk(t):
            for f in fnames:
                if f.endswith(('.tsx','.ts','.jsx','.js','.json')):
                    files.append(os.path.join(root, f))
    for path in files:
        try:
            with open(path, 'r', encoding='utf-8', errors='ignore') as fh:
                lines = fh.readlines()
        except Exception as e:
            continue
        for i, line in enumerate(lines, 1):
            for m in PAT.finditer(line):
                text = m.group(1)
                try:
                    text = text.encode().decode('unicode_escape')
                except Exception:
                    pass
                if is_english_text(text):
                    ctx_start = max(0, i-3)
                    ctx_end = min(len(lines), i+2)
                    ctx = ''.join(lines[ctx_start:ctx_end]).strip()
                    results.append({
                        'file': path,
                        'line': i,
                        'text': text,
                        'context': ctx
                    })

print(json.dumps(results, ensure_ascii=False, indent=2))

File: scripts/find_english_texts.js
const fs = require('fs');
const path = require('path');

const PAT = /["']((?:[^"'\\]|\\.){3,250})["']/g;
const RUSSIAN = /[\u0400-\u04FF]/;

function isEnglishText(s) {
  s = s.trim();
  if (s.length < 3) return false;
  if (!/[A-Za-z]/.test(s)) return false;
  if (RUSSIAN.test(s)) return false;
  if (/^\$\{.*\}$|^\w+$|^[A-Za-z0-9_\-\.\/:@#\u0026\[\]\(\)\|]+$/.test(s)) return false;
  return true;
}

const targets = [
  path.resolve("D:\\АГЕНТ\\ДЖАРВИС\\src\\components\\repo"),
  path.resolve("D:\\АГЕНТ\\ДЖАРВИС\\src\\components\\os"),
  path.resolve("D:\\АГЕНТ\\ДЖАРВИС\\src\\components\\layout"),
  path.resolve("D:\\АГЕНТ\\ДЖАРВИС\\src\\components\\projects"),
  path.resolve("D:\\АГЕНТ\\ДЖАРВИС\\src\\components\\sidebar"),
  path.resolve("D:\\АГЕНТ\\ДЖАРВИС\\src\\components\\settings"),
  path.resolve("D:\\АГЕНТ\\ДЖАРВИС\\src\\app\\repos"),
  path.resolve("D:\\АГЕНТ\\ДЖАРВИС\\src\\app\\page.tsx"),
  path.resolve("D:\\АГЕНТ\\ДЖАРВИС\\src\\lib"),
  path.resolve("D:\\АГЕНТ\\ДЖАРВИС\\src\\services"),
];

const results = [];
for (const t of targets) {
  const files = [];
  if (fs.statSync(t).isFile()) {
    files.push(t);
  } else {
    for (const root of fs.readdirSync(t, { withFileTypes: true, recursive: true })) {
      if (!root.isFile()) continue;
      if (/\.(tsx|ts|jsx|js|json)$/.test(root.name)) {
        files.push(path.join(root.path || root.parentPath, root.name));
      }
    }
  }
  for (const file of files) {
    const lines = fs.readFileSync(file, 'utf-8').split(/\r?\n/);
    for (let i = 0; i < lines.length; i++) {
      const line = lines[i];
      let m;
      while ((m = PAT.exec(line)) !== null) {
        let text = m[1].replace(/\\(.)/g, '$1');
        if (isEnglishText(text)) {
          const ctxStart = Math.max(0, i - 2);
          const ctxEnd = Math.min(lines.length, i + 3);
          const ctx = lines.slice(ctxStart, ctxEnd).join('\n');
          results.push({ file, line: i + 1, text, context: ctx });
        }
      }
    }
  }
}

console.log(JSON.stringify(results, null, 2));
